opendmarc: Free open source software implementation of the DMARC specification

diff --git a/pkgs/development/libraries/opendmarc/default.nix b/pkgs/development/libraries/opendmarc/default.nix
new file mode 100644
index 00000000000..6ad881d9695
--- /dev/null
+++ b/pkgs/development/libraries/opendmarc/default.nix
@@ -0,0 +1,24 @@
+{ stdenv, fetchurl, libbsd, libmilter, libspf2 }:
+
+stdenv.mkDerivation rec {
+  name = "opendmarc-${version}";
+  version = "1.3.2";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/opendmarc/files/${name}.tar.gz";
+    sha256 = "1yrggj8yq0915y2i34gfz2xpl1w2lgb1vggp67rwspgzm40lng11";
+  };
+
+  configureFlags= [
+    "--with-milter=${libmilter}"
+    "--with-spf=${libspf2}"
+  ];
+
+  buildInputs = [ libbsd libmilter libspf2 ];
+
+  meta = with stdenv.lib; {
+    description = "Free open source software implementation of the DMARC specification";
+    homepage = http://www.trusteddomain.org/opendmarc/;
+    platforms = platforms.unix;
+  };
+}
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index db2425e9446..22d82b44200 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3667,6 +3667,8 @@ with pkgs;
 
   opendkim = callPackage ../development/libraries/opendkim { };
 
+  opendmarc = callPackage ../development/libraries/opendmarc { };
+
   opendylan = callPackage ../development/compilers/opendylan {
     opendylan-bootstrap = opendylan_bin;
   };
